### PR TITLE
ref: use guard-collection

### PR DESCRIPTION
**Pull request by xiaoshihou514**
- tests have been adjusted
- checkhealth now only checks for executables since `conf_not_found` is handled in filetype.lua
- updated docs

### CODE REVIEW

File: lua/guard/filetype.lua [hunk 1]
```diff
@@ -1,4 +1,20 @@
 local M = {}
+local ok = #vim.api.nvim_get_runtime_file('lua/guard-collection', false) > 0
```
Comment (glepnir) [inline lua/guard/filetype.lua:2]: no need these.  when is string just check the `guard-collection/formatter`  is exist by using pcall and then check it in table.

File: lua/guard/filetype.lua [hunk 2]
```diff
@@ -1,43 +1,61 @@
...
+  if not pcall(require, 'guard-collection.' .. tool_type) then
+    vim.notify(
+      ('[Guard]: "%s": needs nvimdev/guard-collection to access builtin configuration'):format(
+        tool_name
+      ),
+      4
+    )
+    return
+  end
+  local tbl = require('guard-collection.' .. tool_type)[tool_name]
```
Comment (BBboy01) [inline lua/guard/filetype.lua:13]:
```suggestion
  local is_success, tool = pcall(require, 'guard-collection.' .. tool_type)
  if not is_success then
    vim.notify(
      ('[Guard]: "%s": needs nvimdev/guard-collection to access builtin configuration'):format(
        tool_name
      ),
      4
    )
    return
  end
  local tbl = tool[tool_name]
```
You don't have to `require` twice

File: lua/guard/filetype.lua [hunk 3]
```diff
@@ -1,43 +1,61 @@
...
   local current
   local tbl = {}
   tbl.__index = tbl
 
   function tbl:fmt(config)
     vim.validate({
-      config = { config, { 't', 's' } },
+      config = { config, { 'table', 'string' } },
     })
+    current = 'format'
```
Comment (BBboy01) [inline lua/guard/filetype.lua:33]: There are two lint keys `lint` and `linter`, but there format keys are `fmt`, `format` and `formatter`, perhaps unify the keys of format is a better way

Comment (xiaoshihou514) [reply lua/guard/filetype.lua:33]: I too find this kind of awkward 🤔

**Comment by xiaoshihou514** [comment]
@glepnir Is this pr ready?

**Pull request merged**